### ISSUE TITLE
AI 학습 방향 고민

### ISSUE THREAD

**Issue by SAgiKPJH** [SAgiKPJH/Profile#5]
# AI공부 방향
- 기초 구축 (뉴런 + 신경망 구축 with c++, Python, cuDnn)
- 다양한 인공지능 활용 (YOLO, GAN, Tensorfolw, Keras, Pytorch)
- 기초 다양한 기법 활용 (SVM, 지도학습 등등)
- 인공지능 툴 활용
- 수학 인공지능 (논문)
- c++ cudnn
- python cudnn

# 뉴럴 네트워크
- 근본
  - 핵심은 뉴럴 네트워크를 어떻게 구성하였는가
  - 목적을 위해서 어떤 수학/알고리즘/네트워크구조를 설계하였는가
- 네트워크를 구성하는 방법
  - 순수 네트워크 구축 (기계학습)
  - cuDNN
  - Tensorflow (Keras)
  - Pytorch
- 다른 것은 어떻게 인공지능 툴을 만들었는가?
  - Yolo Darknet
  - Webui
  - 기타 등등
- 기타 인공지능
  - 지도학습, SVM 등등 

# 각각 공부 방법 (해야할 것 + 할 수 있는 것)
- 순수 네트워크 구축해보기
  - [신경망이란?](https://www.youtube.com/playlist?list=PLZHQObOWTQDNU6R1_67000Dx_ZCJB-3pi)
- cuDNN
  - [cuDNN 라이브러리의 Convolution 사용하기](https://m.blog.naver.com/sogangori/220965933190)
  - [cuDNN 으로 컨볼루션 뉴럴 넷 구성하기](https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=sogangori&logNo=220976878680)
- tensorflow (*.ckpt)
  - [TensorFlow Core 튜토리얼](https://www.tensorflow.org/tutorials?hl=ko)
    - [텐서플로 2.0 시작하기: 초보자를 위한 빠른 시작](https://www.tensorflow.org/tutorials/quickstart/beginner?hl=ko)
    - [텐서플로 2.0 시작하기: 전문가를 위한 빠른 시작](https://www.tensorflow.org/tutorials/quickstart/advanced?hl=ko)
    - [Keras를 사용한 ML 기본사항](https://www.tensorflow.org/tutorials/keras/classification?hl=ko)
    - [데이터 로드](https://www.tensorflow.org/tutorials/load_data/csv?hl=ko)
    - [맞춤 설정](https://www.tensorflow.org/tutorials/customization/basics?hl=ko)
    - [분산형 학습](https://www.tensorflow.org/tutorials/distribute/keras?hl=ko)
    - [이미지]
    - [텍스트]
    - [오디오]
    - [구조화된 데이터]
    - [생성]
      - Neural Style Transfer
      - DeepDream
      - 심층 합성곱 생성적 적대 신경망 (DCGANs)
      - Pix2Pix
      - CycleGAN
      - FGSM
      - Autoencoder
      - Variational Autoencoder
    - [모델 이해]
    - [강화학습]
    - [tf.estimator]
    - [어텐션을 사용한 인공 신경망 기계 번역](https://www.tensorflow.org/tutorials/text/nmt_with_attention?hl=ko)
  - [Probability](https://www.tensorflow.org/probability/examples/A_Tour_of_TensorFlow_Probability?hl=ko)
  - [Tendorflow PlayGround](https://playground.tensorflow.org/?hl=ko#activation=tanh&batchSize=10&dataset=circle%C2%AEDataset=reg-plane&learningRate=0.03%C2%AEularizationRate=0&noise=0&networkShape=4,2&seed=0.04620&showTestData=false&discretize=false&percTrainData=50&x=true&y=true&xTimesY=false&xSquared=false&ySquared=false&cosX=false&sinX=false&cosY=false&sinY=false&collectStats=false&problem=classification&initZero=false&hideText=false)
  - [tensorflow App](https://www.tensorflow.org/lite?hl=ko)
  - [TFX](https://www.tensorflow.org/tfx?hl=ko)
- [pytorch](https://tutorials.pytorch.kr/)
  - [TORCHVISION 객체 검출 미세조정(FINETUNING) 튜토리얼](https://tutorials.pytorch.kr/intermediate/torchvision_tutorial.html)
  - [컴퓨터 비전(VISION)을 위한 전이학습(TRANSFER LEARNING)](https://tutorials.pytorch.kr/beginner/transfer_learning_tutorial.html)
- [Keras](https://keras.io/about/)
- yolo darknet 분석 (cuDNN 기반, *.data, *.cfg, *.weights)
- webui 분석 (PyTorch 기반, *.pt)
- c++에서 할 것이냐, python에서 할 것이냐 그것이 문제로다.
- [Apache MXNet](https://mxnet.apache.org/versions/1.4.1/tutorials/index.html)
- [Deeplearning4j]
  - [간단 가이드](https://recordsoflife.tistory.com/219)
